vars/tdr.groovy

```diff
@@ -42,6 +42,22 @@ def getAccountNumberFromStage(String stage) {
   return stageToAccountMap.get(stage)
 }
 
+def commitAndPushGitHubBranch(String branch, String commitMessage) {
+  sshagent(['github-jenkins']) {
+    sh "git config --global user.email tna-digital-archiving-jenkins@nationalarchives.gov.uk"
+    sh "git config --global user.name tna-digital-archiving-jenkins"
+    sh "git add ."
```

Comment:
Would it be possible to limit this to just the relevant files? My worry is that we might accidentally push something else that's been generated by the Jenkins build, just in case it includes sensitive information.

It looks like you're creating the branch in the client build anyway. What about adding and committing in the build, which is going to look different for every project, and then commonising just the push step?

Reply:
So, in the client build it would be something like this:
```
...
steps {
   checkout scm	            
   sh "git checkout ${versionBumpBranch}"
   ... [library version bump code] ...
  sh "git add [files updated]"
  sh "git commit -m [commit message]
  script {
    tdr.pushBranch(branchName)
  }
...
```
And in the library function it would be something like this:
```
...
def pushBranch(String branchName) {
  sh "git push -u origin ${branch}"
}
...
```

Reply:
Yeah, that's what I was thinking.

Ah, does this make it harder to configure the GitHub username and email, because it has to be done before the `git commit` step?

Reply:
yes

Reply:
Changes made